micrometer-api/src/main/java/io/micrometer/api/instrument/MeterRegistry.java

```diff
@@ -92,10 +96,9 @@
 
     private final ObservationConfig observationConfig = new ObservationConfig();
 
-    @Nullable
     @Override
-    public Observation getCurrentObservation() {
-        return localObservation.get();
+    public Optional<Observation> getCurrentObservation() {
```

Comment:
This is an important difference, I changed `MeterRegistry::getCurrentObservation` so that it returns an `Optional` instead of being `@Nullable`.

Reply:
What's the motivation? Won't it make more garbage while callers still have to deal with the absent case?

Reply:
Simply convenience, this can be helpful dealing with the case when the current observation is not available. If you prefer `@Nullable` instead, I'm totally fine rolling this back.